Cmdstanr not forwarding compile errors

I'm not sure if this is me or not, but the develop cmdstanr isn't forwarding me my error messages.

If I compile a model with a syntax error:
```
parameters {
  row_vector[3] a;
  vector[3] b;
}

model {
  matrix[1, 1] c = a * b;
}
```

With this R code:
```
library(cmdstanr)

model = cmdstan_model("/home/bbales2/tmp/test.stan")
```

I'm getting weird errors (instead of the syntax errors I expect):
```
> model = cmdstan_model("/home/bbales2/tmp/test.stan")
Compiling Stan program...
Error in processx::run(command = make_cmd(), args = c(tmp_exe, cpp_options_to_compile_flags(cpp_options),  : 
  System command error
```

This is much better imo:

```
Compiling Stan program...
/
Semantic error in '/tmp/RtmpSiPdjL/model-692018715e71.stan', line 7, column 2 to column 25:
   -------------------------------------------------
     5:  
     6:  model {
     7:    matrix[1, 1] c = a * b;
           ^
     8:  }
   -------------------------------------------------

Ill-typed arguments supplied to assignment operator =: lhs has type matrix and rhs has type real

make: *** No rule to make target '/tmp/RtmpSiPdjL/model-692018715e71.hpp', needed by '/tmp/RtmpSiPdjL/model-692018715e71'.  Stop.
Error in self$compile(...) : 
  An error occured during compilation! See the message above for more information.
```

Without the E> and the rest.

I dont like the "Semantic error in '/tmp/RtmpSiPdjL/model-692018715e71.stan'" and have an idea on how to fix this. I think rstan also compiles in tmp folders and I find it slightly annoying. Opening a PR in a sec.

For what it's worth, on my mac I get this: 

```
Compiling Stan program...
Error in processx::run(command = make_cmd(), args = c(tmp_exe, cpp_options_to_compile_flags(cpp_options),  : 
  System command 'make' failed, exit status: 2, stderr (last 10 lines):
E>      6:  
E>      7:  model {
E>      8:    matrix[1, 1] c = a * b;
E>            ^
E>      9:  }
E>    -------------------------------------------------
E> 
E> Ill-typed arguments supplied to assignment operator =: lhs has type matrix and rhs has type real
E> 
E> make: *** No rule to make target `/var/folders/h6/14xy_35x4wd2tz542dn0qhtc0000gn/T/RtmpzY8D5F/model-11a99479cfaaf.d', needed by `/var/folders/h6/14xy_35x4wd2tz542dn0qhtc0000gn/T/RtmpzY8D5F/model-11a99479cfaaf'.  Stop.
```

which seems to be correct. But I agree the message would look better without the `processx` info at the top, without the `E>`, and without the `make` stuff.

--- a/.github/workflows/R-CMD-check.yaml
+++ b/.github/workflows/R-CMD-check.yaml
@@ -23,7 +23,7 @@ jobs:
           - {os: macOS-latest,   r: 'release'}
           - {os: windows-latest, r: 'devel'}
           - {os: windows-latest, r: 'release'}
-          - {os: windows-latest,   r: 'oldrel'}
+          # - {os: windows-latest,   r: 'oldrel'}
           - {os: ubuntu-16.04,   r: 'release'}
           - {os: ubuntu-16.04,   r: 'oldrel'}
           #- {os: ubuntu-16.04,   r: '3.5'}

--- a/R/model.R
+++ b/R/model.R
@@ -345,7 +345,8 @@ compile_method <- function(quiet = TRUE,
     error_on_status = FALSE
   )
   if (run_log$status != 0) {
-    stop("An error occured during compilation! See the message above for more information.")
+    stop("An error occured during compilation! See the message above for more information.",
+         call. = FALSE)
   }
 
   file.copy(tmp_exe, exe, overwrite = TRUE)

--- a/R/model.R
+++ b/R/model.R
@@ -341,7 +341,9 @@ compile_method <- function(quiet = TRUE,
     echo_cmd = !quiet,
     echo = !quiet,
     spinner = quiet && interactive(),
-    stderr_line_callback = function(x,p) { message(x) },
+    stderr_line_callback = function(x,p) {
+        if (!startsWith(x, paste0(make_cmd(), ": *** No rule to make target"))) message(x)
+    },
     error_on_status = FALSE
   )
   if (run_log$status != 0) {

--- a/R/model.R
+++ b/R/model.R
@@ -340,7 +340,7 @@ compile_method <- function(quiet = TRUE,
     wd = cmdstan_path(),
     echo_cmd = !quiet,
     echo = !quiet,
-    spinner = quiet,
+    spinner = quiet && interactive(),
     stderr_line_callback = function(x,p) { message(x) },
     error_on_status = FALSE
   )

--- a/R/model.R
+++ b/R/model.R
@@ -341,9 +341,12 @@ compile_method <- function(quiet = TRUE,
     echo_cmd = !quiet,
     echo = !quiet,
     spinner = quiet,
-    stderr_line_callback = function(x,p) { if (!quiet) message(x) },
-    error_on_status = TRUE
+    stderr_line_callback = function(x,p) { message(x) },
+    error_on_status = FALSE
   )
+  if (run_log$status != 0) {
+    stop("An error occured during compilation! See the message above for more information.")
+  }
 
   file.copy(tmp_exe, exe, overwrite = TRUE)
   private$cpp_options_ <- cpp_options

--- a/tests/testthat/resources/stan/fail.stan
+++ b/tests/testthat/resources/stan/fail.stan
@@ -1,0 +1,8 @@
+parameters {
+  row_vector[3] a;
+  vector[3] b;
+}
+
+model {
+  matrix[1, 1] c = a * b;
+}

--- a/tests/testthat/test-model-compile.R
+++ b/tests/testthat/test-model-compile.R
@@ -161,3 +161,12 @@ test_that("switching threads on and off works without rebuild", {
   time_diff <- as.double((after_mtime - before_mtime), units = "secs")
   expect_gt(time_diff, 0)
 })
+
+test_that("compile errors are shown", {
+  skip_on_cran()
+  stan_file <- testing_stan_file("fail")
+  expect_error(
+    cmdstan_model(stan_file),
+    "An error occured during compilation! See the message above for more information."
+  )
+})

